packages/jellyfish-api-core/__tests__/category/poolpair/poolswap.test.ts

```diff
@@ -0,0 +1,235 @@
+import { MasterNodeRegTestContainer } from '@defichain/testcontainers'
+import { ContainerAdapterClient } from '../../container_adapter_client'
+import BigNumber from 'bignumber.js'
+import {
+  addPoolLiquidity,
+  createPoolPair,
+  createToken,
+  getNewAddress,
+  mintTokens,
+  utxosToAccount
+} from '@defichain/testing'
+import { RpcApiError } from '../../../src'
+import { PoolSwapMetadata, PoolPairInfo } from '@defichain/jellyfish-api-core/category/poolpair'
+
+describe('poolSwap', () => {
+  const container = new MasterNodeRegTestContainer()
+  const client = new ContainerAdapterClient(container)
+
+  beforeAll(async () => {
+    await container.start()
+    await container.waitForReady()
+    await container.waitForWalletCoinbaseMaturity()
+  })
+
+  afterAll(async () => {
+    await container.stop()
+  })
+
+  it('should poolSwap', async () => {
+    const addressReceiver = await getNewAddress(container)
+    const tokenAddress = await getNewAddress(container)
+    const dfiAddress = await getNewAddress(container)
+    const poolLiquidityAddress = await getNewAddress(container)
+
+    await createToken(container, 'CAT', { collateralAddress: tokenAddress })
+    await utxosToAccount(container, 1200, { address: dfiAddress })
+    await mintTokens(container, 'CAT', { address: dfiAddress })
+    await createPoolPair(container, 'CAT', 'DFI')
+
+    await addPoolLiquidity(container, {
+      tokenA: 'CAT',
+      amountA: 1000,
+      tokenB: 'DFI',
+      amountB: 500,
+      shareAddress: poolLiquidityAddress
+    })
+
+    const metadata: PoolSwapMetadata = {
+      from: dfiAddress,
+      tokenFrom: 'DFI',
+      amountFrom: 555,
+      to: addressReceiver,
+      tokenTo: 'CAT'
+    }
+
+    const poolpairResultBefore = Object.values(await container.call('getpoolpair', ['CAT-DFI']))[0] as PoolPairInfo
+
+    const hex = await client.poolpair.poolSwap(metadata)
+    expect(typeof hex).toStrictEqual('string')
+    expect(hex.length).toStrictEqual(64)
+
+    await container.generate(1)
+
+    const poolpairResultAfter = Object.values(await container.call('getpoolpair', ['CAT-DFI']))[0] as PoolPairInfo
+    const reserveBAfter = new BigNumber(poolpairResultBefore.reserveB).plus(555)
+    const reserveAAfter = new BigNumber(poolpairResultBefore.totalLiquidity).pow(2).div(reserveBAfter) // sqrt(a * b)^2/b
+
+    expect(poolpairResultAfter.reserveB.toFixed(0)).toStrictEqual(reserveBAfter.toFixed(0))
+    expect(poolpairResultAfter.reserveA.toFixed(0)).toStrictEqual(reserveAAfter.toFixed(0))
```

Comment:
does `reserveB` need `toFixed()`?
please use the `toFixed()` properly
and please add the exact result amount at the end of code, easier for reviewer
```
...code.... // {amount}
```

Reply:
Got it.